streaming/src/main/scala/org/apache/spark/streaming/util/RecurringTimer.scala

```diff
@@ -99,6 +99,8 @@ class RecurringTimer(clock: Clock, period: Long, callback: (Long) => Unit, name:
         nextTime += period
         logDebug("Callback for " + name + " called at time " + prevTime)
       }
+      clock.waitTillTime(nextTime)
```

Comment:
Yes, the RecurringTimer is designed to call the function at particular time interval, it should not deviate from that even for the last call.